Bar Cutting Demo (#589)
Avaiga/taipy-doc: docs/knowledge_base/demos/bar_cutting.md

Reviewer: May be a better name would be: 
"Optimizing Bar Cut sizes".
What do you think guys ?

Reviewer 2: I like having the word 'optimize'

Author: This will then appear on the Title of the demo. @jrobinAV you're the one who will add it then, right?

Reviewer: yep!

Reviewer: Done

This visualization is crucial in industries with significant material 
costs or shortages.

[Try it Live](https://bar-cutting.taipy.cloud/Databases)

# Understand the Application
The application comprises five pages accessible via the left panel by 
expanding the menu list.

## Page 1: Bar Visualization

  1. Load the selected dataset Choose from two cases and RUN.
  2. Execute either the Baseline or the Optim algorithms models.
  3. Display results, highlighting loss in blue on a bar chart.
  4. Switch algorithms to compare material waste.

<img src= bar-cutting-bar-Visualization] width="615">

To access additional information, you can open the Parameters panel 
(Taipy Expandable) to review the input data related to the selected case, 
Stock available and Demand to fulfill.

## Page 2: Visual Metrics
On this page, you'll find metrics for both the Baseline Model and the 
Optim Model presented in two distinct formats: bar charts and pie charts. 
These metrics provide insights into the loss, which can be visualized as 
either a percentage of the total length of the mother bars utilized or as 
the precise loss value measured in millimeters.

> <img src=bar-cutting-Metrics width ="615">

Reviewer: TODO JRM: Replace images.

Precision bar cutting is an essential task in various industries, and the Bar Cutting demo 
powered by Taipy is here to transform it. In this demo, we will explore this application's 
capabilities, which optimize bar cutting using different algorithms for enhanced efficiency and 
minimized losses.

# Understanding Bar Cutting Optimization

[Bar Cutting](https://en.wikipedia.org/wiki/Cutting_stock_problem) optimization is the process 
of choosing the best size of cuts to make in bars to minimize waste, and maximize the raw material utilization. It's a critical task in industries where material lost represents important costs or shortfalls.

# The Bar Cutting application

## Page 1: A Glimpse of the Initial Page

> On the initial page of the application, users are presented with the option to select one of the 
> two cases they wish to execute. Clicking the RUN button triggers the following sequence of actions:

Reviewer 3: ```suggestion
On the application's starting page, users can choose one of the two cases they want to run. 
When they click the RUN button, it initiates the following series of actions:
```

1. Loading of the selected dataset (Case 1/Case 2 or Case 3).
2. Execution of two algorithms: Baseline and Optim.
3. Display of results, with the loss depicted in blue on the bar chart.

> The bar chart represents the cutting patterns, showcasing the lengths of customer bars 
> and, notably, the loss. By switching between algorithms, users can instantly observe the 
> difference in loss, with the Optim algorithm typically outperforming the Baseline.

Reviewer 3: ```suggestion
The bar chart displays the cutting patterns, showing the lengths of customer bars and, 
importantly, the amount of material wasted. Users can switch between algorithms to 
quickly see the difference in material waste, with the Optim algorithm usually 
performing better than the Baseline.
```

## Page 2: Exploring Input Data and Metrics

> For deeper insights, users can open the Parameters panel to examine the input 
> data for the chosen case. This includes information about available stock (mother bars for 
> cutting) and demand.

Reviewer 3: ```suggestion
For more detailed information, users can open the Parameters panel to review the input data 
for the selected case. This includes details about the available stock (the original bars for cutting) and the demand.
```

> The Metrics page displays key metrics for both the Baseline and Optim models, offering different 
> representations such as bar charts and pie charts. Metrics showcase the loss as a percentage of 
> the overall length used (mother bars) or as the actual loss value (in millimeters).

Reviewer 3: ```suggestion
The Metrics page displays important metrics for both the Baseline and Optim models, 
presenting them in various ways like bar charts and pie charts. These metrics demonstrate 
the material waste as a percentage of the total length of the original bars (mother bars) used 
or as the actual amount of material wasted (in millimeters).
```

## Page 3: Comparing Model Performance

> The balance icon on the left panel leads users to another page where the performance of the two 
> models is displayed side by side. This comparative view provides valuable insights into which 
> algorithm yields superior results for the selected case.

Reviewer 3: ```suggestion
By clicking the balance icon in the left panel, users are taken to another page where 
the performance of the two models is shown side by side. This comparative view offers valuable insights 
into which algorithm produces better results for the chosen case.
```

## Page 4: Comparing Scenarios 

> For a comprehensive analysis, users can compare the performance of each model across different 
> use cases (scenarios) by clicking on the balance icon. This feature aids in understanding how 
> each algorithm performs under varying conditions.

Reviewer 3: ```suggestion
To conduct a thorough analysis, users can compare how each model performs in various use cases (scenarios) 
by clicking on the balance icon. This feature helps in understanding how each algorithm behaves 
under different conditions.
```

## Page 5: Databases 

> The results are summarized and displayed in a table that can be conveniently downloaded 
> as a CSV file for further analysis.

Reviewer 3: ```suggestion
The results are summarized and presented in a table, which users can easily download as a CSV file for further analysis.
```

# Disclaimer

Please note that the code for this demo is not publicly available at this stage but can be 
shared upon request. Feel free to [contact us](https://www.taipy.io/contact-us/) if you wish to 
receive a sample.

# Conclusion

> The Bar Cutting demo exemplifies Taipy's prowess in precision cutting optimization. Whether 
> you're involved in manufacturing, logistics, or any industry requiring efficient bar cutting, 
> Taipy simplifies the process and minimizes losses in an intuitive and visually appealing manner.

Reviewer 3: ```suggestion
The Bar Cutting demo showcases Taipy's excellence in precision cutting optimization. 
Whether you work in manufacturing, logistics, or any industry that needs efficient bar cutting, 
Taipy makes the process straightforward and reduces waste in a user-friendly and visually appealing way.
```

> Explore the application, test different scenarios, and harness the power of Taipy for more 
> precise and efficient bar cutting.

Reviewer 3: ```suggestion
Take the time to explore the application, experiment with various scenarios, 
and utilize the capabilities of Taipy for more accurate and efficient bar cutting.
```

> Precision bar cutting is an essential task in various industries, and the Bar Cutting demo 

Reviewer: The term "Precision" is wrong to me.
"Precision bar cutting" means being precise when cutting a bar, which is not at all the purpose  of the demo.
The demo defines the length of the pieces to cut in a bar. Nothing related to precision.

Author: @jrobinAV would it work if we make an intro like this: 
Welcome to the Bar Cutting Optimization demonstration, where we showcase an innovative approach to enhance efficiency and minimize losses in the bar-cutting process. This demo will highlight how Taipy simplifies and improves the bar-cutting process, leading to better outcomes with reduced waste.

Reviewer: I would try to be more generic saying Taipy can be used to create decision support applications.
That's kind of a keyword to me.
What about the following?

"Welcome to the Bar Cutting Optimization demonstration, where we showcase an innovative approach to minimize losses in the bar-cutting process. This demo will highlight how a Taipy decision-support application optimize the bar-cutting process, leading to better outcomes with reduced waste."

powered by Taipy is here to transform it. In this demo, we will explore this application's 
capabilities, which optimize bar cutting using different algorithms for enhanced efficiency and 
minimized losses.

# Understanding Bar Cutting Optimization

> [Bar Cutting](https://en.wikipedia.org/wiki/Cutting_stock_problem) optimization is the process 
> of efficiently cutting bars or materials to specific lengths, minimizing waste, and maximizing 
> the utilization of raw materials. It's a critical task in industries where precision and 
> efficiency are paramount.

Reviewer: This is incorrect to me.
```suggestion
[Bar Cutting](https://en.wikipedia.org/wiki/Cutting_stock_problem) optimization is the process 
of choosing the best size of cuts to make in bars to minimize waste, and maximize the raw material utilization. It's a critical task in industries where material lost represents important costs or shortfalls.
```

> # The 5-page application

Reviewer: ```suggestion
# The Bar Cutting application
```

Precision bar cutting is an essential task in various industries, and the Bar Cutting demo 
powered by Taipy is here to transform it. In this demo, we will explore this application's 
capabilities, which optimize bar cutting using different algorithms for enhanced efficiency and 
minimized losses.

# Understanding Bar Cutting Optimization

[Bar Cutting](https://en.wikipedia.org/wiki/Cutting_stock_problem) optimization is the process 

> of choosing the best size of cuts to make in bars to minimize waste, and maximize the raw material utilization. It's a critical task in industries where material lost represents important costs or shortfalls.

Reviewer: Please format the file with line of 100 characters max.

```suggestion
of choosing the best size of cuts to make in bars to minimize waste, and maximize the raw
material utilization. It's a critical task in industries where material lost represents
important costs or shortfalls.
```

<img src=https://github.com/Avaiga/taipy-doc/assets/31435778/7ab84665-083e-4c43-8ccb-c22ed43f3f27 width="500">

## Page 4: Comparing Scenarios 

To conduct a thorough analysis, users can compare how each model performs in various use cases (scenarios) 
by clicking on the balance icon. This feature helps in understanding how each algorithm behaves 
under different conditions.

<img src=https://github.com/Avaiga/taipy-doc/assets/31435778/f9a7c59a-598f-42dd-98dd-63f0cd79ac07 width="500">

## Page 5: Databases 
The results are summarized and presented in a table, which users can easily download as a CSV file for further analysis.

<img src=https://github.com/Avaiga/taipy-doc/assets/31435778/0a1d793b-6e81-4231-8291-9585ff4c8f7b width="500">

# Disclaimer

Please note that the code for this demo is not publicly available at this stage but can be 
shared upon request. Feel free to [contact us](https://www.taipy.io/contact-us/) if you wish to 
receive a sample.

# Conclusion

The Bar Cutting demo showcases Taipy's excellence in precision cutting optimization. 
Whether you work in manufacturing, logistics, or any industry that needs efficient bar cutting, 

> Taipy makes the process straightforward and reduces waste in a user-friendly and visually appealing way.

Reviewer: We missed to change this paragraph. We do not deal with cut precision.

Take the time to explore the application, experiment with various scenarios, 

> and utilize the capabilities of Taipy for more accurate and efficient bar cutting.

Reviewer: Same here. We don't deal with cutting accuracy or efficiency. We deal with optimizing cut sizes. Not the cutting process.
